core/src/main/scala/org/apache/spark/storage/BlockManagerMasterActor.scala

```diff
@@ -522,7 +523,9 @@ private[spark] class BlockManagerInfo(
         logInfo("Removed %s on %s on tachyon (size: %s)".format(
           blockId, blockManagerId.hostPort, Utils.bytesToString(blockStatus.tachyonSize)))
       }
+      return BlockStatus(storageLevel, 0, 0, 0)
     }
+    null
```

Comment:
you don't need `return` here -- the last value of each block is its return value.  so this could be:

```
if (storageLevel.isValid) {
 ...
  _blocks.get(blockId)
} else if (_blocks.containsKey(blockId)) {
  ...
  BlockStatus(storageLevel, 0, 0, 0)
} else {
  null
}
```